.github/workflows/steam.yml

```diff
@@ -0,0 +1,214 @@
+
+    - name: Determine if Steam upload should run
+      # If the nightly build has already been uploaded and thus a cache exists skip this and the following steps.
+      # Steam does not prevent us from uploading duplicate builds so this would just pollute the dashboard.
+      # This is a bit of a hack and can fail to work if our cache has been evicted or we somehow have no commits for 7 days,
+      # but it's better than nothing!
+      id: should-run
+      run: |
+        if [[ '${{ env.UPLOAD_TYPE }}' == 'release' || '${{ steps.cache.outputs.cache-hit }}' != 'true' ]]; then
+            echo "::set-output name=result::true"
+        else
+            echo "::set-output name=result::false"
+        fi
+
+    - name: Download and prepare builds
+      if: steps.should-run.outputs.result == 'true'
+      run: |
+        echo "::group::Download Windows build"
+        if [[ '${{ env.WIN_ARTIFACT_URL }}' == *'api.github.com'* ]]; then
+            curl -L -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' '${{ env.WIN_ARTIFACT_URL }}' -o windows.zip
+        else
+            curl -L '${{ env.WIN_ARTIFACT_URL }}' -o windows.zip
+        fi
+        echo "::endgroup::"
+      
+        echo "::group::Download Mac build"
+        if [[ '${{ env.MAC_x86_ARTIFACT_URL }}' == *'api.github.com'* ]]; then
+            curl -L -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' '${{ env.MAC_x86_ARTIFACT_URL }}' -o mac_x86.dmg.zip
+        else
+            curl -L '${{ env.MAC_x86_ARTIFACT_URL }}' -o mac_x86.dmg
+        fi
+        echo "::endgroup::"
+
+        mkdir -p steam && cd steam
+  
+        echo "::group::Extract and prepare Win64"
+        mkdir steam-windows
+        (
+            cd steam-windows
+            unzip ../../windows.zip
+            # CI builds can be double-zipped
+            if compgen -G "*.zip" > /dev/null; then
+                unzip *.zip
+                rm *.zip
+            fi
+            # copy install scripts and create sentinel file
+            cp -r ../../source/CI/steam/scripts scripts
+            touch disable_updater
+        )
+        echo "::endgroup::"
+
+        echo "::group::Extract macOS (x86)"
+        mkdir steam-macos
+        # CI builds are zipped
+        if [[ -f ../mac_x86.dmg.zip ]]; then
+            unzip ../mac_x86.dmg.zip
+            # 7-Zip will have an exit code of 2 due to the "unsafe" 'Applications' symlink.
+            # GitHub treats this as a failure so ignore non-zero exit codes here.
+            7zz x *.dmg -otmp || true
+        else
+            7zz x ../mac_x86.dmg -otmp || true
+        fi
+          
+        mv tmp/*/OBS.app steam-macos
+        echo "::endgroup::"
+
+    - name: Setup steamcmd
+      if: steps.should-run.outputs.result == 'true'
+      uses: CyberAndrii/setup-steamcmd@v1
+
+    - name: Generate Steam auth code
+      if: steps.should-run.outputs.result == 'true'
+      id: steam-totp
+      uses: CyberAndrii/steam-totp@v1
```

Comment:
This is just my personal choice, but when it comes to Marketplace actions that handle secret tokens, I rather specify the commit hash that I reviewed as tags can be changed after the fact and can point to unreviewed commits without anyone noticing.

Reply:
Good point, I'll use the hashes for the steamcmd/totp action.